src/huggingface_hub/hf_api.py

```diff
@@ -80,7 +81,7 @@ def _validate_repo_id_deprecation(repo_id, name, organization):
     return name, organization
 
 
-def repo_type_and_id_from_hf_id(hf_id: str):
+def repo_type_and_id_from_hf_id(hf_id: str, _hf_co_domain: Optional[str] = None):
```

Comment:
why a private-ish input arg? We can have it public, and maybe call it `hub_url`? Also needs to be documented in the docstring.